IfBuildProfile doesn't work properly with messaging

### Describe the bug

Create a bean which uses e.g. `@Incoming` and annotate it with `@IfBuildProfile`. Then run it without the build profile.

Quarkus now complains about missing configuration if the channels, and still activates them.

### Expected behavior

The bean should be completely disabled. No  warnings, no MQTT connection.

### Actual behavior

The bean is somewhat active. Prices (from the example) don't get generated, but the MQTT connection is active.

### How to Reproduce?

Take a look at the reproducer: https://github.com/ctron/quarkus-repro-1

In a nutshell:

* Annotate all beans with `@IfBuildProfile("foo")`
* Then run without the profile `foo`

### Output of `uname -a` or `ver`

Linux xxx 6.0.12-300.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Dec 8 16:58:47 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.14.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)

### Additional information

_No response_

With 

```java
@IfBuildProfile("foo")
@ApplicationScoped
public class PriceConverter {

    private static final double CONVERSION_RATE = 0.88;

    @Incoming("prices")
    @Outgoing("my-data-stream")
    @Broadcast
    public double process(byte[] priceRaw) {
        int priceInUsd = Integer.parseInt(new String(priceRaw));
        System.out.println("Receiving price: " + priceInUsd);
        return priceInUsd * CONVERSION_RATE;
    }

}
```

and 

```properties
# Configure the MQTT sink (we write to it)
mp.messaging.outgoing.topic-price.type=smallrye-mqtt
mp.messaging.outgoing.topic-price.topic=prices
mp.messaging.outgoing.topic-price.host=localhost
mp.messaging.outgoing.topic-price.port=1883
mp.messaging.outgoing.topic-price.auto-generated-client-id=true

# Configure the MQTT source (we read from it)
mp.messaging.incoming.prices.type=smallrye-mqtt
mp.messaging.incoming.prices.topic=prices
mp.messaging.incoming.prices.host=localhost
mp.messaging.incoming.prices.port=1883
mp.messaging.incoming.prices.auto-generated-client-id=true
```

at runtime we get:

```
2022-12-13 17:56:19,751 WARN  [io.sma.rea.mes.provider] (Quarkus Main Thread) SRMSG00207: Some components are not connected to either downstream consumers or upstream producers:
	- IncomingConnector{channel:'prices', attribute:'mp.messaging.incoming.prices'} has no downstream
	- OutgoingConnector{channel:'topic-price', attribute:'mp.messaging.outgoing.topic-price'} has no upstream

2022-12-13 17:56:19,770 ERROR [io.ver.mqt.imp.MqttClientImpl] (vert.x-eventloop-thread-0) Can't connect to localhost:1883: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:1883
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1589)
```

--- a/docs/src/main/asciidoc/amqp-reference.adoc
+++ b/docs/src/main/asciidoc/amqp-reference.adoc
@@ -817,3 +817,18 @@ Type: _string_ | false |
 Type: _string_ | false |
 
 |===
+
+=== Conditionally configure channels
+
+You can configure the channels using a specific profile.
+Thus, the channels are only configured (and added to the application) when the specified profile is enabled.
+
+To achieve this, you need:
+
+1. Prefix the `mp.messaging.[incoming|outgoing].$channel` entries with `%my-profile` such as `%my-profile.mp.messaging.[incoming|outgoing].$channel.key=value`
+2. Use the `@IfBuildProfile("my-profile")` on the CDI beans containing `@Incoming(channel)` and `@Outgoing(channel)` annotations that need only to be enabled when the profile is enabled.
+
+Note that reactive messaging verifies that the graph is complete.
+So, when using such a conditional configuration, ensure the application works with and without the profile enabled.
+
+Note that this approach can also be used to change the channel configuration based on a profile.

--- a/docs/src/main/asciidoc/kafka.adoc
+++ b/docs/src/main/asciidoc/kafka.adoc
@@ -2320,6 +2320,21 @@ Attribute values are resolved as follows:
 2. if not set, the connector looks for a `Map` with the channel name or the configured `kafka-configuration` (if set) and the value is retrieved from that `Map`
 3. If the resolved `Map` does not contain the value the default `Map` is used (exposed with the `default-kafka-broker` name)
 
+=== Conditionally configure channels
+
+You can configure the channels using a specific profile.
+Thus, the channels are only configured (and added to the application) when the specified profile is enabled.
+
+To achieve this, you need:
+
+1. Prefix the `mp.messaging.[incoming|outgoing].$channel` entries with `%my-profile` such as `%my-profile.mp.messaging.[incoming|outgoing].$channel.key=value`
+2. Use the `@IfBuildProfile("my-profile")` on the CDI beans containing `@Incoming(channel)` and `@Outgoing(channel)` annotations that need only to be enabled when the profile is enabled.
+
+Note that reactive messaging verifies that the graph is complete.
+So, when using such a conditional configuration, ensure the application works with and without the profile enabled.
+
+Note that this approach can also be used to change the channel configuration based on a profile.
+
 == Integrating with Kafka - Common patterns
 
 === Writing to Kafka from an HTTP endpoint

